sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -716,6 +776,197 @@ cdef class Splitter:
+                    break
+
+                gain = _split_gain(sum_gradient_left, sum_hessian_left,
+                                    sum_gradient_right, sum_hessian_right,
+                                    loss_current_node, monotonic_cst,
+                                    lower_bound, upper_bound,
+                                    self.l2_regularization)
+                if gain > best_gain and gain > self.min_gain_to_split:
+                    found_better_split = True
+                    best_gain = gain
+                    best_direction = direction
+                    best_sorted_thres = sorted_idx
+                    best_sum_gradient_left = sum_gradient_left
+                    best_sum_hessian_left = sum_hessian_left
+                    best_n_samples_left = n_samples_left
+
+
+        if found_better_split:
+            split_info.gain = best_gain
+
+            # bin_idx is unused for categorical splits: cat_bitset is used
+            # instead and set below
+            split_info.bin_idx = 0
+
+            split_info.sum_gradient_left = best_sum_gradient_left
+            split_info.sum_gradient_right = sum_gradients - best_sum_gradient_left
+            split_info.sum_hessian_left = best_sum_hessian_left
+            split_info.sum_hessian_right = sum_hessians - best_sum_hessian_left
+            split_info.n_samples_left = best_n_samples_left
+            split_info.n_samples_right = n_samples - best_n_samples_left
+
+            # We recompute best values here but it's cheap
+            split_info.value_left = compute_node_value(
+                split_info.sum_gradient_left, split_info.sum_hessian_left,
+                lower_bound, upper_bound, self.l2_regularization)
+
+            split_info.value_right = compute_node_value(
+                split_info.sum_gradient_right, split_info.sum_hessian_right,
+                lower_bound, upper_bound, self.l2_regularization)
+
+            # create bitset with values from best_sorted_thres
+            init_bitset(split_info.cat_bitset)
+
+            if best_direction == 1:  # left
+                for i in range(best_sorted_thres + 1):
+                    bin_idx = cat_sorted_infos[i].bin_idx
+                    set_bitset(split_info.cat_bitset, bin_idx)
+            else:
+                for i in range(best_sorted_thres + 1):
+                    bin_idx = cat_sorted_infos[n_used_bin - 1 - i].bin_idx
+                    set_bitset(split_info.cat_bitset, bin_idx)
+
+            if has_missing_values:
+                split_info.missing_go_to_left = in_bitset(
+                    split_info.cat_bitset, missing_values_bin_idx)
+
+        free(cat_sorted_infos)
+
+
+cdef int compare_cat_infos(const void * a, const void * b) nogil:
+    cdef:
+        categorical_info a_info = (<categorical_info *>a)[0]
+        categorical_info b_info = (<categorical_info *>b)[0]
+    return 1 if a_info.value <= b_info.value else -1
```

Comment:
```py
1 if a_info.value <= b_info.value else -1
```

This means that the array will be sorted in decreasing order. Why is this needed? Let's document this please :)

Reply:
Updated with -1 at the front and with being strictly less than i.e:

```
-1 if a_info.value < b_info.value else 1
```